### ISSUE TITLE
backward through ME.SparseTensor

### ISSUE THREAD

**Issue by hua-wu-que** [StanfordVL/MinkowskiEngine#148]
Hi there,

I have a network G that generate a feature vector "feats" of size [b,n] together with a position vector "coords" of size [n, 3]*b (a list of b array of size [n,3])
I then create a sparse tensor using 
``
        sinput = ME.SparseTensor(
            feats=feats.cpu(),
            coords=ME.utils.batched_coordinates(coords),
            quantization_mode=ME.SparseTensorQuantizationMode.UNWEIGHTED_AVERAGE
        ).to(feats)
``
Next, I feed sinput into a MinkowskiConvolution layer  defined as 
``
        self.enc_block_s1 = nn.Sequential(
            ME.MinkowskiConvolution(
                ch[0], ch[1], kernel_size=3, stride=1, dimension=3),
            ME.MinkowskiBatchNorm(ch[1]),
            ME.MinkowskiELU(),
        ).cuda()

loss = self.enc_block(feat).F.mean().backward()
``
But calling loss.backward() cannot backpropage gradient to network G, (printing G's parameters gradient got None).

could you give me some suggestions on how to do this backward?

Many thanks!

**Comment by hua-wu-que** [comment]
Hi @chrischoy ,

To make the question more clear, I am wondering how to make the sparse tensor creation step differentiable in MinkowskiEngine.

I saw most of the example code, you are creating a sparse tensor 'sin' during data loader phase, then pass it through the later network. i.e. in such case, you don't need to backpropagate gradient to 'sin'

Thank you!

**Comment by chrischoy** [comment]
you are applying backward twice.

**Comment by yxmt-oss** [comment]
We have also encountered the same gradient vanishing issue. We hope to obtain the gradient information of coordinates after backpropagation through the model. Have you solved this problem？
